pkg/storage/leveldb_backend.go

```diff
@@ -63,120 +63,100 @@ func newLevelDBBackend(
 	if err != nil {
 		return nil, err
 	}
-	regionStorageCtx, regionStorageCancel := context.WithCancel(ctx)
 	lb := &levelDBBackend{
-		StorageEndpoint:     endpoint.NewStorageEndpoint(levelDB, ekm),
-		ekm:                 ekm,
-		batchSize:           defaultBatchSize,
-		flushRate:           defaultFlushRegionRate,
-		batchRegions:        make(map[string]*metapb.Region, defaultBatchSize),
-		flushTime:           time.Now().Add(defaultFlushRegionRate),
-		regionStorageCtx:    regionStorageCtx,
-		regionStorageCancel: regionStorageCancel,
+		StorageEndpoint: endpoint.NewStorageEndpoint(levelDB, ekm),
+		ekm:             ekm,
+		batchSize:       defaultBatchSize,
+		flushRate:       defaultFlushRate,
+		batch:           make(map[string][]byte, defaultBatchSize),
+		flushTime:       time.Now().Add(defaultFlushRate),
 	}
+	lb.ctx, lb.cancel = context.WithCancel(ctx)
 	go lb.backgroundFlush()
 	return lb, nil
 }
 
-var dirtyFlushTick = time.Second
-
 func (lb *levelDBBackend) backgroundFlush() {
 	defer logutil.LogPanic()
 
 	var (
 		isFlush bool
 		err     error
 	)
-	ticker := time.NewTicker(dirtyFlushTick)
+	ticker := time.NewTicker(defaultDirtyFlushTick)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
 			lb.mu.RLock()
 			isFlush = lb.flushTime.Before(time.Now())
-			failpoint.Inject("regionStorageFastFlush", func() {
+			failpoint.Inject("levelDBStorageFastFlush", func() {
 				isFlush = true
 			})
 			lb.mu.RUnlock()
 			if !isFlush {
 				continue
 			}
 			if err = lb.Flush(); err != nil {
-				log.Error("flush regions meet error", errs.ZapError(err))
+				log.Error("flush data meet error", errs.ZapError(err))
 			}
-		case <-lb.regionStorageCtx.Done():
+		case <-lb.ctx.Done():
 			return
 		}
 	}
 }
 
-func (lb *levelDBBackend) SaveRegion(region *metapb.Region) error {
-	region, err := encryption.EncryptRegion(region, lb.ekm)
-	if err != nil {
-		return err
-	}
+// SaveInBatch saves the key-value pair in the batch cache, and it will
+// only be saved to the underlying storage when the `Flush` method is
+// called or the cache is full.
+func (lb *levelDBBackend) SaveInBatch(key string, value []byte) error {
```

Comment:
Better call it `Save`? This method saves only one pair of key values

Reply:
I changed it to `SaveIntoBatch` to emphasize the use of batch cache. PTAL.